src/http-api/errors.ts

```diff
@@ -28,11 +29,13 @@ interface IErrorJson extends Partial<IUsageLimit> {
  * specific to HTTP responses.
  * @param msg - The error message to include.
  * @param httpStatus - The HTTP response status code.
+ * @param httpHeaders - The HTTP response headers.
  */
 export class HTTPError extends Error {
     public constructor(
         msg: string,
         public readonly httpStatus?: number,
+        public readonly httpHeaders?: Headers,
```

Comment:
afaict this is only used in a couple of places, and `httpStatus` and `httpHeaders` are set in both. Do we really need to make them optional?

Reply:
Since this is an exported class, I left this optional so as to not break API compatibility.

On that note, it's worth considering an API version bump to make them mandatory (as well as `errcode` and `error` in `MatrixError`, which the spec says are mandatory).

Reply:
I don't think it's correct that this class is exported outside the js-sdk?

Reply:
oh, cancel that, it does seem to be. Ugh.

Reply:
I would vote for an API version bump. We don't want to have to be coding around it not being populated forever.

Reply:
For the time being I left this as it is, as optional values don't hinder much & are still used internally.  With that said, I'm in favour of a dedicated PR for making them mandatory.